capstone/src/main/java/com/google/sps/search/Trie.java

```diff
@@ -93,6 +96,140 @@ public void insert(String name, String fullName) {
     return names;
   }
 
+  /**
+   * Checks whether the input can be split into a full name by splitting input at each index and
+   * checking whether the newly created first and last name exist in the set of full names in the
+   * trie. For example, input "andrewsweet" can be split in "Andrew Sweet," and if this name is in
+   * the trie, then the name will be added to the set. Boolean reversed indicates whether to search
+   * for the name but first and last name are reversed. For example, "sweetandrew" is reversed and
+   * it can be split but into "Sweet Andrew," for which we can check that the reverse ("Andrew
+   * Sweet") is in the trie.
+   *
+   * @param input user input
+   * @param reversed boolean representing whether name is reversed
+   * @return Set of name suggestions
+   */
+  public Set<String> whitespace(String input, boolean reversed) {
+    Set<String> fullNames = new TreeSet<String>();
+
+    for (int i = 0; i < input.length(); i++) {
+      String firstName = input.substring(0, i);
+      String lastName = input.substring(i);
+      String fullName = "";
+      if (reversed) {
+        fullName = lastName + " " + firstName;
+      } else {
+        fullName = firstName + " " + lastName;
+      }
+      if (this.hasFullName(firstName, fullName)) {
+        fullNames.add(fullName);
+      }
+    }
+    return fullNames;
+  }
+
+  /**
+   * Finds names that are within the maximum LED distance of input by calling a recursive helper
+   * method that uses dynamic programming to find the edit distance between words.
+   *
+   * @param input user input
+   * @return Map from possible names to its edit distance from input
+   */
+  public Map<String, Integer> searchLed(String input) {
+    input = input.toUpperCase();
+    Map<String, Integer> suggestionsMap = new HashMap<String, Integer>();
```

Comment:
Either move this closer to the first usage of this variable, or remove it entirely, instead returning and merging suggestionsMaps, recursively.